pkg/internal/apply/layerApplier.go

```diff
@@ -154,6 +155,51 @@ func (a KubectlLayerApplier) decodeAddons(layer layers.Layer,
 	}
 }
 
+func (a KubectlLayerApplier) addOwnerRefs(layer layers.Layer, hrs []*helmopv1.HelmRelease) error {
+	for i, hr := range hrs {
+		a.logDebug("Adding owner ref to HelmRelease for AddonsLayer", layer, "index", i, "helmRelease", hr)
+		err := controllerutil.SetControllerReference(layer.GetAddonsLayer(), hr, a.scheme)
+		if err != nil {
+			// could not apply owner ref for object
+			return fmt.Errorf("Unable to apply owner reference for AddonsLayer '%s' to HelmRelease '%s'", layer.GetName(), getLabel(hr))
```

Comment:
Needto wrap err here as the real error message is never reported.

Reply:
Agreed.  There's probably a linter rule for this!

Reply:
@richardcase - I've applied this change.  Please review and resolve.